### PR TITLE
Add new lint rule for hardcoded coroutine dispatchers

### PR DESCRIPTION
**Pull request by CDRussell**
Task/Issue URL: https://app.asana.com/0/488551667048375/1202882025878544/f

### Steps to test this PR

_Feature 1_
- [ ]
- [ ]

### UI changes
| Before  | After |
| ------ | ----- |
!(Upload before screenshot)|(Upload after screenshot)|

### CODE REVIEW

**Comment by CDRussell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #2440** <a href="https://app.graphite.dev/github/pr/duckduckgo/android/2440" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/android/2440?utm_source=stack-comment).

File: lint-rules/build.gradle [hunk 1]
```diff
@@ -34,4 +34,10 @@ java {
 
 tasks.register('jvm_checks') {
     dependsOn 'test'
+}
+
+jar {
```
Comment (malmstein) [inline lint-rules/build.gradle:39]: why this change? I don't see any description about it. Does this still make the `com.duckduckgo.lint.ui.DuckDuckGoDesignSystemRegistry` work?

Comment (CDRussell) [reply lint-rules/build.gradle:39]: It's still WIP so incomplete. Design registry not wired in yet. 

I couldn't find the right syntax to add both registries in. It might be possible with the right syntax, or it might be you can only register one and we'd have to combine the registries. Not sure yet. 

Let's discuss more why the PR at all next week, but TL;DR is that our custom rules aren't running at the moment; only their tests execute but not the lint rules themselves.

**Comment by CDRussell** [comment]
Closing; work carried on in https://github.com/duckduckgo/Android/pull/2446 instead
